### PR TITLE
Extract FactCard component from Cause Page Stats

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request extracts a new utility `FactCard` component from the rendered 'stats' on the Cause Page. Also adds documentation and a test assertion.

### How should this be reviewed?
~Does the new name make sense? @lkpttn did not want to commit, but I feel like he liked referring to these as Facts and not Stats~ (https://github.com/DoSomething/phoenix-next/pull/2187#issuecomment-637069025)

[👁️ on the review app](https://dosomething-phoenix-de-pr-2187.herokuapp.com/us/causes/education)

### Any background context you want to provide?
The dream was always to have these lil factoid cards globally available. We do intent to support these in Collection Pages as well so this felt like the perfect time to extract it into a component.

### Relevant tickets

References [Pivotal #172697820](https://www.pivotaltracker.com/story/show/172697820).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [x] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens. - _No Changes!_
- [x] Added appropriate feature/unit tests.

### CODE REVIEW

**Comment by aaronschachter** [comment]
> Does the new name make sense? @lkpttn did not want to commit, but I feel like he liked referring to these as Facts and not Stats

This one feels pretty subjective.. but I think I'd prefer `StatCard` because `FactCard` immediately makes me think this would live on a 11 facts page (which I just learned today by searching through the repo isn't even a content type anymore 👴 ). But I do think this looks like what I was getting at in https://www.pivotaltracker.com/story/show/172729110/comments/214586558, for a BigBangBlock kind of a component)

**Comment by mendelB** [comment]
> > Does the new name make sense? @lkpttn did not want to commit, but I feel like he liked referring to these as Facts and not Stats
> 
> This one feels pretty subjective.. but I think I'd prefer `StatCard` because `FactCard` immediately makes me think this would live on a 11 facts page (which I just learned today by searching through the repo isn't even a content type anymore 👴 ). But I do think this looks like what I was getting at in https://www.pivotaltracker.com/story/show/172729110/comments/214586558, for a BigBangBlock kind of a component)

Ah that's a perfectly compelling enough reason to go with 'Stat', will update! 

This does feel a touch bigger? than those examples since it's more stylized, in its own card & background styling, but (and maybe this was precisely your point), I wonder if it can't at least _utilize_ a formalized 'Action Stat/Item Block' or whatnot.

File: docs/development/features/stat-card.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Stat Card
+
+The `StatCard` is a UI utility component which renders stat information in a card with a customizable background color.
+
+## Usage Instructions
+
+Render the `StatCard` component with the following props:
+
+- `title`: the title of the card, formatted to all caps
+- `number`: the stat number whichs gets formatted to a U.S. English locale (comma separated)
```
Comment (aaronschachter) [inline docs/development/features/stat-card.md:10]: `whichs` 🥪

Comment (mendelB) [reply docs/development/features/stat-card.md:10]: I invent the words as I go

**Pull request merged**